src/collect_columns/collect_columns.py

```diff
@@ -46,24 +47,20 @@ def collect_columns(count_tables: List[Path], feature_column: int,
     # If no sample names are given use table basenames instead.
     if names is None:
         names = [path.name for path in count_tables]
-    # Convert tables_have_headers to conform with the pandas parameter.
-    header = 0 if tables_have_headers else None
-    # Load in the first table.
-    merged_table = pd.read_csv(count_tables.pop(0), sep=sep, header=header,
-                               index_col=feature_column,
-                               usecols=[feature_column, value_column])
-    merged_table.columns = [names.pop(0)]
-    merged_table.index.names = ["feature"]
-    # Merge the other tables into the first.
+    counts = {}
```

Comment:
Nitpick: You can also use the `dict()` constructor here as it is slightly more explicit.

Reply:
And to comment on my own comment. The constructor is much faster. In applications like this that is preferred.